src/classes/UTIL_DMLService_TEST.cls

```diff
@@ -0,0 +1,697 @@
+
+        Test.startTest();
+        UTIL_DMLService.upsertRecord(con, false);
+        Test.stopTest();
+
+        List<Contact> retrievedUpsertedContacts = [SELECT Id, FirstName, LastName, AccountId
+                                                   FROM Contact];
+
+        System.assertEquals(false, retrievedUpsertedContacts.isEmpty());
+
+        con.FirstName = 'Cloudy';
+        UTIL_DMLService.upsertRecord(con, false);
+
+        retrievedUpsertedContacts = [SELECT Id, FirstName, LastName, AccountId
+                                     FROM Contact];
+
+        System.assertEquals(con.FirstName, retrievedUpsertedContacts[0].FirstName);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests upsert of a List of records with specified allOrNone value.
+    * verify:
+    *   Records upserted
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRecordsUpsertWithAllOrNone() {
+        if (strTestOnly != '*' && strTestOnly != 'testRecordsUpsertWithAllOrNone') return;
+
+        Account acct = new Account(Name = 'UpsertedAccount');
+        insert acct;
+
+        List<Contact> contacts = new List<Contact> { new Contact(FirstName = 'Blue', LastName = 'Cloud', AccountId = acct.Id),
+                                                     new Contact(FirstName = 'Astro', LastName = 'Cloud', AccountId = acct.Id) };
+
+        Test.startTest();
+        UTIL_DMLService.upsertRecords(contacts, false);
+        Test.stopTest();
+
+        List<Contact> retrievedUpsertedContacts = [SELECT Id, FirstName, LastName, AccountId
+                                                   FROM Contact];
+
+        System.assertEquals(false, retrievedUpsertedContacts.isEmpty());
+
+        contacts[0].FirstName = 'Cloudy';
+        contacts[0].FirstName = 'Cody';
+        UTIL_DMLService.upsertRecords(contacts, false);
+
+        retrievedUpsertedContacts = [SELECT Id, FirstName, LastName, AccountId
+                                     FROM Contact];
+
+        System.assertEquals(contacts[0].FirstName, retrievedUpsertedContacts[0].FirstName);
+        System.assertEquals(contacts[1].FirstName, retrievedUpsertedContacts[1].FirstName);
+    }
+
+
+    /*********************************************************************************************************
+    * @description Tests the NPSP Setting Respect Duplicate Rule Settings for duplicate Accounts
+    *   set to false 
+    * verify:
+    *   Duplicate Accounts detected and allowed when the Respect Duplicate Rule Setting set to true.
+    * @return void
+    **********************************************************************************************************/
+    private static testMethod void testRespectDuplicateRuleSettingsForAccounts() {
```

Comment:
it would be good to also get in testing that covers the new upsert methods.  rather than copying this whole test and using upsert instead of update, maybe make a common routine that takes a parameter on whether to do upsert vs update, and then have to thin testmethod covers that use it, one doing update, one doing upsert.